prov/util/src/util_coll.c

```diff
@@ -1178,31 +1178,37 @@ int ofi_av_set(struct fid_av *av, struct fi_av_set_attr *attr,
 {
 	struct util_av *util_av = container_of(av, struct util_av, av_fid);
 	struct util_av_set *av_set;
-	int ret, iter;
+	uint64_t offset;
+	int ret;
 
 	if (!util_av->coll_mc) {
 		ret = util_coll_av_init(util_av);
 		if (ret)
 			return ret;
 	}
 
-	av_set = calloc(1,sizeof(*av_set));
+	av_set = calloc(1, sizeof(*av_set));
 	if (!av_set)
 		return -FI_ENOMEM;
 
 	ret = util_av_set_init(av_set, util_av, context);
 	if (ret)
 		goto err1;
 
-	av_set->fi_addr_array = calloc(ofi_av_size(util_av),
+	size_t max_size = attr->count;
```

Comment:
Please declare all variables at the top of the functions and not in the middle.  Finding declarations mixed throughout code is difficult on someone trying to read/maintain the code.

Reply:
Sorry; fixed.